core/src/main/java/org/apache/iceberg/MetricsUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+public class MetricsUtil {
+
+  private MetricsUtil() {
+  }
+
+  public static Map<Integer, Long> getNanValueCounts(
```

Comment:
nit: seems like iceberg prefers method names without `get`.

Reply:
I was aware of that, but couldn't come up with a good name without `get` in this case (things like `generateCount`/`createCount` is longer and sounds like just for workaround the word "get"), and later assume it was fine since it's not an ordinary getter... Do you have a recommendation?

Reply:
The problem with `get` in this case is that it isn't clear where the return value is coming from. I think that `create` is a better option because it is clear that the return value is built from the input arguments.

An example of where the value may come from somewhere else is `IPUtil.getHostName(String iface)`. The input value is used, but the actual return value would come from an external source.

Reply:
Makes sense, I'll update in both here and #1829